test/StateSchedulingTest.cpp

```diff
@@ -0,0 +1,125 @@
+#include "pipert/Scheduler.h"
+#include "pipert/ScheduledChannel.h"
+#include "pipert/PolledChannel.h"
+#include "pipert/Packet.h"
+#include "pipert/Timer.h"
+#include "gtest/gtest.h"
+#include <iostream>
+#include <unistd.h>
+
+namespace {
+
+class Base {
+ public:
+  int GetValue() const { return value_; }
+ protected:
+  int value_ = 0;
+};
+
+class Processor : public Base {
+ public:
+  void Process(::pipert::PacketToProcess<int> packet) {
+    for (int i = 0; i < 10; ++i) {
+      value_ += packet.data();
+    }
+    for (int i = 0; i < 10; ++i) {
+      value_ -= packet.data();
+    }
+  }
+};
+
+class Incrementor : public Base {
+ public:
+  Incrementor(::pipert::ScheduledChannel<int>&& decChannel) : decChannel_(std::move(decChannel)) {}
+  void Increment(::pipert::PacketToProcess<int> packet) {
+    value_ += packet.data();
+    ::pipert::PacketToFill<int> new_packet = decChannel_.Acquire("Decrementor", packet.timestamp(), packet.data());
+    while(packet.IsEmpty()) {
+      new_packet = decChannel_.Acquire("Decrementor", packet.timestamp(), packet.data());
+    }
```

Comment:
This part seems sort of fishy. Are you sure that it's `packet`, and not `new_packet` that should be tested here?

Reply:
Thanks, you are right, the ```new_packet``` emptiness should be tested.